ConfigSpace/hyperparameters.pyx

```diff
@@ -1395,6 +1677,304 @@ cdef class NormalIntegerHyperparameter(IntegerHyperparameter):
+        cdef int sampled_neighbors = 0
+        _neighbors_as_int = set()  # type: Set[int]
+        cdef long long int_value = self._transform(value)
+        cdef long long new_int_value = 0
+        cdef float new_value = 0.0
+        cdef np.ndarray samples
+        cdef double[:] samples_view
+        
+        norm_std = std * (self._upper - self._lower) 
+        if self.upper - self.lower <= n_requested:
+            transformed_value = self._transform(value)
+            for n in range(self.lower, self.upper + 1):
+                if n != int_value:
+                    if transform:
+                        neighbors.append(n)
+                    else:
+                        n = self._inverse_transform(n)
+                        neighbors.append(n)
+
+        else:
+            samples = rs.normal(loc=value, scale=norm_std, size=250)
+            samples_view = samples
+
+            while sampled_neighbors < n_requested:
+
+                while True:
+                    new_value = samples_view[idx]
+                    idx += 1
+                    i += 1
+                    if idx >= 250:
+                        samples = rs.normal(loc=value, scale=norm_std, size=250)
+                        samples_view = samples
+                        idx = 0
+                    if new_value < self._lower or new_value > self._upper:
+                        continue
+                    new_int_value = self._transform(new_value)
+                    if int_value == new_int_value:
+                        continue
+                    elif i >= 200:
+                        # Fallback to uniform sampling if generating samples correctly
+                        # takes too long
+                        values_to_sample = [j for j in range(self.lower, self.upper + 1)
+                                            if j != int_value]
+                        samples = rs.choice(
+                            values_to_sample,
+                            size=n_requested,
+                            replace=False,
+                        )
+                        for sample in samples:
+                            if transform:
+                                neighbors.append(sample)
+                            else:
+                                sample = self._inverse_transform(sample)
+                                neighbors.append(sample)
+                        break
+                    elif new_int_value in _neighbors_as_int:
+                        continue
+                    elif int_value != new_int_value:
+                        break
+
+                _neighbors_as_int.add(new_int_value)
+                sampled_neighbors += 1
+                if transform:
+                    neighbors.append(new_int_value)
+                else:
+                    new_value = self._inverse_transform(new_int_value)
+                    neighbors.append(new_value)
+
+        return neighbors
```

Comment:
I'm not going to claim any expertise here either. Since the Beta and Uniform are both bounded, I simply grabbed the local search from the respective uniform parameters and rescaled sampling where appropriate, since the Beta is not forced into a [0, 1] domain.

As such, it functions the exact same (or at least, it should) as the Uniform parameters. I'm of course happy to help/iterate/improve the local search, but it made sense to me to just grab what was already there.

Reply:
Ah okay, you can see my knowledge of this library is not that extensive. I will spend some time optimizing this loop and put the code back here.